### PR TITLE
Document overworld.c and link stuff in general

### PR DESCRIPTION
**Pull request by Phlosioneer**
Lots of reverse engineering. Lots of progress.

### CODE REVIEW

**Comment by Phlosioneer** [comment]
Xerneas on the discord wants me to investigate the `npc_something` function / `EVENT_TYPE_ID` / event callbacks more before this is merged.
I'll do a deep dive on the event script system tomorrow.

**Comment by Phlosioneer** [comment]
Alright, I looked through the event scripts more carefully this time. Xerneas's concerns have been addressed, and this PR is ready to be reviewed & merged.

File: src/overworld.c [hunk 1]
```diff
@@ -2975,102 +3037,116 @@ static u8 GetLinkPlayerIdAt(s16 x, s16 y)
...
     struct EventObject *eventObj = &gEventObjects[eventObjId];
 
     if (linkPlayerEventObj->active)
     {
-        if (a2 > 10)
+        if (facing > FACING_FORCED_RIGHT)
             eventObj->triggerGroundEffectsOnMove = 1;
-        else
-            gUnknown_08339E00[gUnknown_08339DC8[linkPlayerEventObj->mode](linkPlayerEventObj, eventObj, a2)](linkPlayerEventObj, eventObj);
+        else {
```
Comment (Diegoisawesome) [inline src/overworld.c]: Balance braces

File: src/overworld.c [hunk 2]
```diff
@@ -2975,102 +3037,116 @@ static u8 GetLinkPlayerIdAt(s16 x, s16 y)
...
 }
 
-static u8 npc_something3(u8 a1, u8 a2)
+// Flip Up/Down facing codes. If newFacing doesn't specify a direction, default
+// to oldFacing. Note that this clears also the "FORCED" part of the facing code,
+// even for Left/Right codes.
+static u8 FlipVerticalAndClearForced(u8 newFacing, u8 oldFacing)
 {
-    switch (a1 - 1)
+    switch (newFacing - 1)
```
Comment (Diegoisawesome) [inline src/overworld.c]: Clean up the switch by removing all the -1s

File: src/record_mixing.c [hunk 3]
```diff
@@ -953,43 +957,47 @@ static void sub_80E7FF8(u8 taskId)
...
             task->data[0] ++;
         }
         break;
     case 5:
-        if (gReceivedRemoteLinkPlayers == 0)
+        if (gReceivedRemoteLinkPlayers == FALSE)
             DestroyTask(taskId);
         break;
+
+    // Mixing Emerald/FireRed/LeafGreen records.
```
Comment (Diegoisawesome) [inline src/record_mixing.c]: FireRed/LeafGreen can't mix records

File: src/script_pokemon_util_80F87D8.c [hunk 4]
```diff
@@ -640,22 +641,24 @@ void ScriptSetMonMoveSlot(u8 monIndex, u16 move, u8 slot)
     SetMonMoveSlot(&gPlayerParty[monIndex], move, slot);
 }
 
-void sub_80F9438(void)
+// Note: When control returns to the event script, gSpecialVar_Result be
```
Comment (Diegoisawesome) [inline src/script_pokemon_util_80F87D8.c]: gSpecialVar_Result **will** be

File: src/overworld.c (hunk 2 above)
Comment (Phlosioneer) [reply src/overworld.c]: But if the -1's are removed, the constants can't be used. The switch operates by subtracting one from everything, shouldn't the cases reflect that?

**Comment by Phlosioneer** [comment]
Ugh. This PR is before I had a sense of the conventions and layout of the Repo, there's so many mistakes and misunderstandings.

File: data/maps/TradeCenter/map.json [hunk 5]
```diff
@@ -46,22 +46,24 @@
   ],
   "coord_events": [
     {
+      "comment": "This is trade seat #1.",
```
Comment (huderlem) [inline data/maps/TradeCenter/map.json]: Let's not add any custom "comment" fields at this time.  The name of the event script should be sufficient.

File: include/save.h [hunk 6]
```diff
@@ -83,13 +83,11 @@ bool8 sub_8153380(void);
 bool8 sub_81533AC(void);
 u8 sub_81533E0(void);
 u8 sub_8153408(void);
-u8 sub_8153430(void);
-bool8 sub_8153474(void);
+u8 FullSaveGame(void); //FullSaveGame
```
Comment (huderlem) [inline include/save.h]: Remove unnecessary comments

File: src/berry_blender.c [hunk 7]
```diff
@@ -146,8 +146,8 @@ extern void sub_81AABF0(void (*callback)(void));
 extern void sub_800B4C0(void);
 extern void ClearLinkCallback(void);
 extern void sub_8009F8C(void);
-extern void sub_8153430(void);
-extern bool8 sub_8153474(void);
+extern void FullSaveGame(void);
```
Comment (huderlem) [inline src/berry_blender.c]: remove these externs and include save.h

File: common_syms/overworld.txt [hunk 8]
```diff
@@ -1,8 +1,8 @@
 gBGTilemapBuffers1
 gBGTilemapBuffers2
 gBGTilemapBuffers3
-gUnknown_03005DA8
+gHeldKeyCodeToSend
 gFieldCallback
 gFieldCallback2
-gUnknown_03005DB4
+LocalLinkPlayerId
```
Comment (huderlem) [inline common_syms/overworld.txt]: Add the `g` prefix back to this global variable

File: src/overworld.c [hunk 9]
```diff
@@ -2192,566 +2219,600 @@ static void sub_8086B14(void)
...
 {
     s32 i;
     for (i = 0; i < 4; i++)
-        sUnknown_03000E10[i] = 0x80;
+        sPlayerTradingStates[i] = PLAYER_TRADING_STATE_IDLE;
 }
 
-static bool32 sub_8086CA8(u16 a1)
+// Returns true if all connected players are in tradingState.
+static bool32 AreAllPlayersInState(u16 tradingState)
```
Comment (huderlem) [inline src/overworld.c]: `AreAllPlayersInTradingState()`, to be consistent

File: src/overworld.c [hunk 10]
```diff
@@ -2192,566 +2219,600 @@ static void sub_8086B14(void)
...
+            if (trainer->isLocalPlayer)
             {
-                c1_link_related_func_set(sub_80870F8);
+                SetKeyInterceptCallback(KeyInterCB_DeferToEventScript);
                 sub_8087530(script);
             }
             return;
         }
-        if (sub_8086CE0(0x83) == 1)
+        if (AreAnyPlayersInTradingState(PLAYER_TRADING_STATE_EXITING_ROOM) == TRUE)
```
Comment (huderlem) [inline src/overworld.c]: `IsAnyPlayerInTradingState()`

File: data/maps/TradeCenter/map.json (hunk 5 above)
Comment (Phlosioneer) [reply data/maps/TradeCenter/map.json]: My bad; that was for personal use while I was reverse-engineering. Removed.

**Comment by Phlosioneer** [comment]
This PR is fixed.

File: data/maps/TradeCenter/map.json [hunk 11]
```diff
@@ -52,16 +52,17 @@
       "elevation": 3,
       "var": "VAR_TEMP_0",
       "var_value": "0",
-      "script": "gUnknown_082773F5"
+      "script": "EventScript_TradeCenter_Chair0"
     },
     {
+      "comment": "This is trade seat #2.",
```
Comment (huderlem) [inline data/maps/TradeCenter/map.json]: you forgot one.

File: include/link.h [hunk 12]
```diff
@@ -299,7 +299,7 @@ extern struct LinkPlayer gLocalLinkPlayer;
 
 bool32 Link_AnyPartnersPlayingRubyOrSapphire(void);
 bool32 sub_800A03C(void);
-void SetLocalLinkPlayerId(u8);
+void SetgLocalLinkPlayerId(u8);
```
Comment (huderlem) [inline include/link.h]: global find/replace accident

**Pull request merged**